app/views/form_drafts/show.haml

```diff
@@ -13,13 +13,16 @@
   = form_tag controller: :forms, action: :submit, id: @draft.form.id do
     = fields_for :user do |u|
       .field
-        .heading Please complete the following personal information.
-        //.explanation You will be able to log in and review your submission.
+        .explanation 
+          This is the personal information we have on file for you.
+          Please fill in any missing or incorrect information.
+          The confirmation will be sent to this email address.
       - [:first_name, :last_name, :email].each do |field|
         .field
           .label= u.label field
-          = u.text_field field, size: 50, required: true,
-            value: @current_user.try(field)
+          = u.text_field field, size: 50,
+            value: @current_user.try(field) || @placeholder.try(field),
```

Comment:
I want to clarify something here.  The Shibboleth attributes aren't headers. The are present in the `env` of *every* request to the site (because of `mod_shibboleth`).  That's why I'm confused about why we're copying them wholesale from one ephemeral data store to another.  It seems like you *can* just use them where they are.